classic/selection/FeatureModel.js

```diff
@@ -234,5 +292,36 @@ Ext.define('GeoExt.selection.FeatureModel', {
         }
 
         me.callParent(arguments);
+    },
+
+    /**
+     * Ovrwrite parent's destroy method in order to unregister the OL events,
+     * that were added on init.
+     * Needed due to the lack of destroy event of the parent class.
+     *
+     * @private
+     */
+    destroy: function() {
+        var me = this;
+
+        // unfortunately callParent triggers the bindComponent method, so we
+        // have to know if we are in the process of destroying or not.
+        me._destroying = true;
+
+        me.unbindOlEvents();
+        me.callParent(arguments);
+
+        me._destroying = false;
+    },
+
+    /**
+     * Returns a random feature ID.
+     *
+     * @private
+     * @return {String} Random feature ID
+     */
+    getRandomFid: function() {
+        // current timestamp plus a random int between 0 and 10
+        return new Date().getTime() + Math.floor(Math.random() * 11);
```

Comment:
This returns a `number` right now. Did you want to write

```js
return new Date().getTime() + '' +  Math.floor(Math.random() * 11);
```

The current implementation can easily produce colliding numbers.

I am a bit tempted to use a utility for this like

```js
// from https://stackoverflow.com/posts/8809472/revisions
function generateUUID() { // Public Domain/MIT
    var d = new Date().getTime();
    if (typeof performance !== 'undefined' && typeof performance.now === 'function'){
        d += performance.now(); //use high-precision timer if available
    }
    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
        var r = (d + Math.random() * 16) % 16 | 0;
        d = Math.floor(d / 16);
        return (c === 'x' ? r : (r & 0x3 | 0x8)).toString(16);
    });
}
```

I'll let you decide since it also feels like a bit overkill.

Reply:
Yes, in general the current implementation can produce duplicate IDs. But in this case I doubt that there will be duplicates. The `getRandomFid` function is triggered by a user click interaction, so the clicker has to be very fast :wink:. I pledge to leave it as it is for the moment. In the future we can refactor this by introducing a util class offering `generateUUID` and use this here.